### PR TITLE
[13.0][MIG] account_asset_management: Migration to 13.0

### PR DESCRIPTION
**Pull request by ernestotejeda**
Waiting for:

- [x] https://github.com/odoo/odoo/pull/44239

Main changes:
- Python code and views were adapted to be compatible with v13
- When assets are created through accounting journal items, they are created when the journal items is posted.
- When a Bill Invoice is created or modified, at the time it is saved, for each line that has a selected asset profile and Quantity *'N'* greater than 1, it will be replaced by *'N'* lines identical to it but with quantity 1. This was done to maintain the same behavior as in the previous version, in which for each asset created there is a Journal Item. In addition, this solution does not change the data model which does not cause migration scripts.
- The configuration option was removed so the only function of that is to allow the module to be uninstalled by unchecking that configuration option
- Tests were adapted

Cc @Tecnativa TT19675

### CODE REVIEW

File: account_asset_management/models/account_asset.py [hunk 1]
```diff
@@ -519,12 +504,73 @@ def open_entries(self):
             "domain": [("id", "in", am_ids)],
         }
 
-    @api.multi
-    def compute_depreciation_board(self):
+    def _group_lines(self, table):
```
Comment (yajo) [inline account_asset_management/models/account_asset.py:507]: Please add a docstring for this method. Its code is not obvious.

File: account_asset_management/models/account_asset.py [hunk 2]
```diff
@@ -519,12 +504,73 @@ def open_entries(self):
...
+
+    def _compute_depreciation_line(
+        self,
+        depreciated_value_posted,
+        table_i_start,
+        line_i_start,
+        table,
+        last_line,
+        posted_lines,
+    ):
```
Comment (yajo) [inline account_asset_management/models/account_asset.py:539]: Add docstring too please.

File: account_asset_management/models/account_asset.py [hunk 3]
```diff
@@ -519,12 +504,73 @@ def open_entries(self):
...
+                        "line_days": line["days"],
+                        "init_entry": entry["init"],
+                    }
+                    depreciated_value += round(amount, digits)
+                    depr_line = self.env["account.asset.line"].create(vals)
+                else:
+                    seq -= 1
+            line_i_start = 0
+
+    def compute_depreciation_board(self):
```
Comment (yajo) [inline account_asset_management/models/account_asset.py:573]: Add docstring too please.

File: account_asset_management/models/account_move.py [hunk 4]
```diff
@@ -60,6 +58,45 @@ def write(self, vals):
                 )
         return super().write(vals)
 
+    def post(self):
+        super().post()
```
Comment (yajo) [inline account_asset_management/models/account_move.py]: Shouldn't you return this at the end?

File: account_asset_management/models/account_move.py [hunk 5]
```diff
@@ -60,6 +58,45 @@ def write(self, vals):
...
+                    vals["company_id"] = self.env.context["company_id"]
+                asset = (
+                    self.env["account.asset"]
+                    .with_context(create_asset_from_move_line=True, move_id=move.id)
+                    .create(vals)
+                )
+                aml.with_context(allow_asset=True).asset_id = asset.id
+
+    def button_draft(self):
+        super().button_draft()
```
Comment (yajo) [inline account_asset_management/models/account_move.py]: Shouldn't you return this at the end?

**Review by yajo — APPROVED** [review]

File: account_asset_management/models/account_move.py (hunk 4 above)
Comment (ernestotejeda) [reply account_asset_management/models/account_move.py]: yes, you're right. Done

File: account_asset_management/models/account_move.py (hunk 5 above)
Comment (ernestotejeda) [reply account_asset_management/models/account_move.py]: yes, you're right, done

File: account_asset_management/models/account_asset.py (hunk 3 above)
Comment (pedrobaeza) [reply account_asset_management/models/account_asset.py:573]: In this case, the method name is self-explanatory: this computes the depreciation board of the asset.

File: account_asset_management/models/account_asset.py (hunk 2 above)
Comment (pedrobaeza) [reply account_asset_management/models/account_asset.py:539]: I think this one is also explanatory: it computes a depreciation line.

File: account_asset_management/models/account_asset.py (hunk 1 above)
Comment (pedrobaeza) [reply account_asset_management/models/account_asset.py:507]: A possible docstring is the one that is as a comment:

```
Group lines prior to depreciation start period.
```

**Review by yajo — APPROVED** [review]

**Comment by ernestotejeda** [comment]
Changes done

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
I have squashed commit history a bit, mostly squashing administrative commits. @ernestotejeda please remember to do it.

Let's proceed with the merge and later PRs on 12.0 will be cherry-picked here, but we have already a very good initial version, considering all the refactoring on removal of the account.invoice model.

/ocabot merge

**Pull request merged**